scripts/nb-tester/test-notebook.py

```diff
@@ -193,7 +192,13 @@ def cancel_trailing_jobs(start_time: datetime) -> bool:
     """
     # QiskitRuntimeService().jobs() includes qiskit-ibm-provider jobs too
     service = QiskitRuntimeService()
-    jobs = [j for j in service.jobs(created_after=start_time) if not j.in_final_state()]
+
+    def _is_not_finished(job):
+        # Force runtime to update job status
+        job.status()
+        return not job.in_final_state()
+
+    jobs = list(filter(_is_not_finished, service.jobs(created_after=start_time)))
```

Comment:
Workaround for https://github.com/Qiskit/qiskit-ibm-runtime/issues/1547

Reply:
It'd be good to link to this in a comment so that we don't keep it forever.

Reply:
```suggestion
    def _is_not_finished(job):
        # Force runtime to update job status
        # Workaround for Qiskit/qiskit-ibm-runtime#1547
        job.status()
        return not job.in_final_state()

    jobs = list(filter(_is_not_finished, service.jobs(created_after=start_time)))
```